ckanext/activity/model/activity.py

```diff
@@ -784,3 +810,26 @@ def _activity_stream_get_filtered_users() -> list[str]:
         users_list = [site_user.get("name")]
 
     return model.User.user_ids_for_name_or_id(users_list)
+
+
+def _filter_activities_by_permission_labels(
+    q: QActivity,
+    user_permission_labels: Optional[list[str]] = None,
+):
+    """Adds a filter to an existing query object to
+    exclude package activities based on user permissions.
+    """
+
+    # `user_permission_labels` is None when user is sysadmin
+    if user_permission_labels:
```

Comment:
```suggestion
    if user_permission_labels is not None:
```
Don't want to confuse an empty list (possible normal user with no permission labels) with None